prac_05/emails.py

```diff
@@ -0,0 +1,28 @@
+"""
+CP1404 - Prac 5
+Program: Emails
+"""
+
+def main():
+    """Create dictionary of emails to names, and displays it."""
+    email_to_name = {}
+    email = input('Email: ')
+
+    while email != "":
+        name = extract_name(email)
+        name_confirmation = input(f"Is your name {name}? (Y/n) ").upper()
+        if name_confirmation != "" and name_confirmation != "Y":
+            name = input("Name: ").title()
+        email_to_name[email] = name
+        email = input('Email: ')
+
+    for email, name in email_to_name.items():
+        print(f"{name} ({email})")
+
+def extract_name(email):
+    """Extract name from email."""
+    unformatted_name = email.split('@')[0]
```

Comment:
This method assumes that the email will always be in a first.last@example.com format. This might not work as expected for other formats.